docs/diagrams/src/user-model-mapping.puml

```diff
@@ -0,0 +1,84 @@
+@startuml
+    entity User {
+        - id: String
+        ..
+        - typeId: UUID
+        - accountType: AccountType
+        --
+        - provider: Provider
+        - emailAddress: String
+    }
+
+    entity Teacher {
+        - id: UUID
+        ..
+        - firstName: String
+        - surname: String
+        - location: Location
+        - locale: String
+        - availability: Set<DayOfWeek>
+    }
+
+    entity School {
+        - id: UUID
+        ..
+        - name: String
+        - contactPersonName: String
+        - location: Location
+        - locale: String
+    }
+
+    entity Location {
+        - address: String
+        - postalCode: String
+        - longitude: Decimal
+        - latitude: Decimal
+    }
+
+    enum Provider {
+        AUTH0
+        --
+        +isAssociatedWith()
+    }
+
+    enum AccountType {
+        SCHOOL
+        TEACHER
```

Comment:
hier zou ik ook ADMIN verwachten; dit benoemen we al op andere plaatsen en dit lijkt mij de plek.

Reply:
Wellicht geen admin rol maar iets van een 'medewerker' / LiTTiL type? Die geef je vervolgens de rol admin. Ik denk niet dat een account type per definitie admin hoeft te zijn. Wel scherp overigens, daar moeten we wel wat mee idd.

Reply:
Of gewoon een OTHER introduceren, want met de juiste rol admin werkzaamheden kan verrichten.

Reply:
klopt; of wellicht dan gewoon nullable maken, maar OTHER kan ook